src/internal/components/Draggable.tsx

```diff
@@ -0,0 +1,84 @@
+import { useCallback, useEffect, useState } from 'react';
+import { cn } from '../../styles/theme';
+
+type DraggableProps = {
+  children: React.ReactNode;
+  gridSize?: number;
+  startingPosition?: { x: number; y: number };
+  enableSnapToGrid?: boolean;
+};
+
+export default function Draggable({
+  children,
+  gridSize = 1,
+  startingPosition = { x: 20, y: 20 },
+  enableSnapToGrid = true,
+}: DraggableProps) {
+  const [position, setPosition] = useState(startingPosition);
+  const [dragOffset, setDragOffset] = useState({ x: 0, y: 0 });
+  const [isDragging, setIsDragging] = useState(false);
+
+  const snapToGrid = useCallback(
+    (positionValue: number) => {
+      return Math.round(positionValue / gridSize) * gridSize;
+    },
+    [gridSize],
+  );
+
+  const handleDragStart = (e: React.PointerEvent) => {
+    setIsDragging(true);
+
+    setDragOffset({
+      x: e.clientX - position.x,
+      y: e.clientY - position.y,
+    });
+  };
+
+  useEffect(() => {
+    if (!isDragging) {
+      return;
+    }
+
+    const handleGlobalMove = (e: PointerEvent) => {
+      setPosition({
+        x: e.clientX - dragOffset.x,
+        y: e.clientY - dragOffset.y,
+      });
+    };
+
+    const handleGlobalEnd = () => {
+      setPosition((prev) => ({
+        x: enableSnapToGrid ? snapToGrid(prev.x) : prev.x,
+        y: enableSnapToGrid ? snapToGrid(prev.y) : prev.y,
+      }));
+      setIsDragging(false);
+    };
+
+    document.addEventListener('pointermove', handleGlobalMove);
+    document.addEventListener('pointerup', handleGlobalEnd);
+
+    return () => {
+      document.removeEventListener('pointermove', handleGlobalMove);
+      document.removeEventListener('pointerup', handleGlobalEnd);
+    };
+  }, [isDragging, dragOffset, snapToGrid, enableSnapToGrid]);
+
+  return (
+    <div
+      data-testid="ockDraggable"
+      className={cn(
+        'fixed select-none',
+        isDragging ? 'cursor-grabbing' : 'cursor-grab',
+      )}
+      style={{
+        left: `${position.x}px`,
+        top: `${position.y}px`,
+        zIndex: 1000,
+        touchAction: 'none',
```

Comment:
these two styles can be classes.  may want to move zIndex to a const file as const zIndex = { modal: 1000 } so we can use element type zindexes across components.

Reply:
with regard to `left` and `top` — these cannot be classes. class-based was my first approach, but I found that since tailwind classes are applied at build-time, they do not recalculate. and since these properties need to be dynamic (they govern in real-time where the draggable component is on the page), inline styles is required here.

to double check, i just tried this, and can confirm that this does not work:
<img width="484" alt="image" src="https://github.com/user-attachments/assets/54a925d6-a9aa-466e-80d9-de1b7ee79b3d" />

Reply:
I like the zIndex const idea though, i've implemented that with placeholder values

Reply:
i meant touchAction: 'none' -> touch-none above

Reply:
ah, sorry, yeah that can be, i will move